x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/10_keyword.yml

```diff
@@ -162,3 +162,22 @@ setup:
                 query: Monday
                 analyzer: standard
   - match: {hits.total.value: 0}
+
+---
+"dynamic mapping update":
+  - do:
+      bulk:
+        index: sensor
+        refresh: true
+        body: |
+          {"index":{}}
+          {"animal": "cow"}
+  - do:
+      search:
+        index: sensor
+        body:
+          query:
+            match:
+              animal: cow
+  - match: {hits.total.value: 1}
+  - match: {hits.hits.0._source.animal: cow}
```

Comment:
I was wondering if it would be possible to have some other way of testing this besides a yaml test, a single node test would also be good I think